### PR TITLE
DOCAPI-7719: Columns matcher docs

### PR DESCRIPTION
**Pull request by BayoNet**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Category (leave one):
- Documentation for #6038

### CODE REVIEW

File: docs/en/query_language/select.md [hunk 1]
```diff
@@ -957,12 +957,31 @@ Running a query may use more memory than 'max_bytes_before_external_sort'. For t
...
 
-The expressions specified in the SELECT clause are analyzed after the calculations for all the clauses listed above are completed.
-More specifically, expressions are analyzed that are above the aggregate functions, if there are any aggregate functions.
-The aggregate functions and everything below them are calculated during aggregation (GROUP BY).
-These expressions work as if they are applied to separate rows in the result.
+[Expressions](syntax.md#syntax-expressions) that specified in the `SELECT` clause are analyzed after the calculations for all the clauses listed above are completed. More specifically, expressions are analyzed that are above the aggregate functions, if there are any aggregate functions. The aggregate functions and everything below them are calculated during aggregation (`GROUP BY`). These expressions work as if they are applied to separate rows in the result.
+
+If you want to get all columns in the result, use the asterisk (`*`) symbol. For example, `SELECT * FROM ...`.
+
+You can also use the `COLUMNS` expression to match some columns in the result by a regular expression.
```
Comment (blinkov) [inline docs/en/query_language/select.md]: which regular expression implementation is used? (all of them have some limitations)

File: docs/en/query_language/select.md [hunk 2]
```diff
@@ -957,12 +957,31 @@ Running a query may use more memory than 'max_bytes_before_external_sort'. For t
...
+
+```sql
+COLUMNS('regexp')
+```
+
+Example:
+
+```sql
+SELECT COLUMNS('n') + COLUMNS('u') FROM system.numbers LIMIT 2
```
Comment (blinkov) [inline docs/en/query_language/select.md]: * what will happen if no columns match?
* what will happen if multiple columns match? does it matter if all have same/compatible type?
* what will happen if we replace `+` with `,` in this example?
* what will be the result of `toTypeName(COLUMNS(...))`?

**Review by blinkov — CHANGES_REQUESTED** [review]

File: docs/en/query_language/select.md (hunk 1 above)
Comment (BayoNet) [reply docs/en/query_language/select.md]: @mfridental Can you answer this question, please?

File: docs/en/query_language/select.md (hunk 2 above)
Comment (BayoNet) [reply docs/en/query_language/select.md]: @mfridental Can you formulate some limitations for using this functionality that can help to answer Ivan's questions?

File: docs/en/query_language/select.md (hunk 1 above)
Comment (alexey-milovidov) [reply docs/en/query_language/select.md]: re2

File: docs/en/query_language/select.md (hunk 2 above)
Comment (alexey-milovidov) [reply docs/en/query_language/select.md]: > what will happen if no columns match?

The function `+` will be called with zero number of arguments and exception will be thrown.

> what will happen if multiple columns match?

The function `+` will be called with more than two arguments and exception will be thrown.

> what will happen if we replace + with , in this example?

Everything gonna be alright.

> what will be the result of toTypeName(COLUMNS(...))

The only valid way is when COLUMNS will match single column and the function `toTypeName` will be called with a single argument.

**Pull request merged**